### PR TITLE
Bolt v3.0: Add Constructible Stylesheets to Web Components

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1786

### CODE REVIEW

File: packages/servers/testing-server/index.js [hunk 1]
```diff
@@ -79,8 +79,8 @@ getConfig().then(async boltConfig => {
             .join('\n')}</style>
 
             ${normalizeAssets(assetsByChunkName['bolt-global'])
-              .filter(path => path.endsWith('.js'))
-              .map(path => `<script src="${path}"></script>`)
+              .filter(path => path.endsWith('.mjs'))
+              .map(path => `<script type="module" src="${path}"></script>`)
```
Comment (sghoweri) [inline packages/servers/testing-server/index.js]: Need to revert these changes after I'm done merging in the latest from master

File: docs-site/src/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+//   for (var i = 0, len = docsSiteLinks.length; i < len; i++) {
+//     const linkElem = docsSiteLinks[i];
+//     const href = linkElem.getAttribute('href');
+//     if (href) {
+//       if (href.startsWith('http')) {
+//         linkElem.setAttribute('target', '_blank');
+//       }
+//     }
+//   }
+// });
```
Comment (adamszalapski) [inline docs-site/src/index.js]: We have here a lot of commented code, we need this code for reference or we can remove it?

File: example-integrations/drupal-lab/package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     }
   ],
   "scripts": {
-    "setup:js": "cd web/themes/bolt-starter && yarn && npm run build",
+    "setup:js": "cd web/themes/bolt-starter && npm install && npm run build",
```
Comment (adamszalapski) [inline example-integrations/drupal-lab/package.json]: In my opinion, is better to use yarn command instead of npm because is faster. The second thing is we using workspaces and that is only working with yarn.

File: jest.config.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const testFilesToIgnore = globby.sync([
   'packages/components/**/*.e2e.js',
   'packages/components/**/*.no2e.js', // workaround to disable a few e2e tests 
   'packages/**/*.data.js',
+  'packages/generators/**/*.test.js',
```
Comment (adamszalapski) [inline jest.config.js]: Good idea i don't know why myself don't do that.

File: packages/build-tools/create-webpack-config.js [hunk 5]
```diff
@@ -471,8 +490,7 @@ async function createWebpackConfig(buildConfig, isModern) {
       quiet: true,
       clientLogLevel: 'none',
       stats: statsPreset(webpackStats[config.verbosity]),
-      hot: false,
-      // hot: config.prod ? false : true,
+      hot: config.prod ? false : true,
```
Comment (adamszalapski) [inline packages/build-tools/create-webpack-config.js]: In my opinion, we should use a sort version like `hot: config.prod,` it will have the same result but with less code :)

File: .travis.yml [hunk 6]
```diff
@@ -65,11 +59,7 @@ jobs:
       script:
         - composer global require hirak/prestissimo
         - cd example-integrations/drupal-lab
-        - composer install --prefer-dist --no-dev
-        - cd web/themes/bolt-starter && yarn --force
-        - yarn run build
-        - yarn run test # run Jest tests on the FE build that just compiled
-        - cd ../../../ && composer run build # finish running through the remainder of the Drupal Lab build
+        - npm run setup
```
Comment (adamszalapski) [inline .travis.yml]: If we install yarn earlier maybe is a good idea to change this to `yarn run setup`

**Comment by sghoweri** [comment]
Closing this one out in lieu of #1526 — appreciate you checking this out @adamszalapski!!
